Engine/BuiltInShaders/common/CelLightSource.sh

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------------------------------------//
+// @brief Calculates the contribution of all light sources.                                //
+//                                                                                         //
+// vec3 CalculateLights(Material material, vec3 worldPos, vec3 viewDir, vec3 diffuseBRDF); //
+//-----------------------------------------------------------------------------------------//
+
+#include "../UniformDefines/U_Light.sh"
+
+uniform vec4 u_lightCountAndStride;
+uniform vec4 u_lightParams[LIGHT_LENGTH];
+
+U_Light GetLightParams(int pointer) {
+	// struct {
+	//   /*0*/ struct { float type; vec3 position; };
+	//   /*1*/ struct { float intensity; vec3 color; };
+	//   /*2*/ struct { float range; vec3 direction; };
+	//   /*3*/ struct { float radius; vec3 up; };
+	//   /*4*/ struct { float width, height, lightAngleScale, lightAngleOffeset; };
+	// }
+	
+	U_Light light;
+	light.type              = u_lightParams[pointer + 0].x;
+	light.position          = u_lightParams[pointer + 0].yzw;
+	light.intensity         = u_lightParams[pointer + 1].x;
+	light.color             = u_lightParams[pointer + 1].yzw;
+	light.range             = u_lightParams[pointer + 2].x;
+	light.direction         = u_lightParams[pointer + 2].yzw;
+	light.radius            = u_lightParams[pointer + 3].x;
+	light.up                = u_lightParams[pointer + 3].yzw;
+	light.width             = u_lightParams[pointer + 4].x;
+	light.height            = u_lightParams[pointer + 4].y;
+	light.lightAngleScale   = u_lightParams[pointer + 4].z;
+	light.lightAngleOffeset = u_lightParams[pointer + 4].w;
+	return light;
+}
+
+// -------------------- point -------------------- //
+//TODO add cartoon point light
+
+// -------------------- Directional -------------------- //
+
+vec3 CalculateDirectionalLight(U_Light light, Material material, vec3 worldNormal, vec3 viewDir, vec3 diffuseBRDF) {
```

Comment:
Is there so many difference between LightSource.sh and CelLightSource.sh that you have to create a new Light Source file, or do you just need to add a new CalculateCelDirectionalLight function in LightSource.sh?